### PR TITLE
Register case_07 in CMakeLists

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: src/libasr/pass/pass_utils.cpp [hunk 1]
```diff
@@ -421,6 +421,9 @@ namespace LCompilers {
                 return ASRUtils::EXPR(ASR::make_RealBinOp_t(al, loc, left, op, right, type, nullptr));
             } else if (ASRUtils::is_complex(*type)) {
                 return ASRUtils::EXPR(ASR::make_ComplexBinOp_t(al, loc, left, op, right, type, nullptr));
+            } else if (ASRUtils::is_character(*type)) {
+                ASR::ttype_t *type_ = ASRUtils::TYPE(ASR::make_Integer_t(al, type->base.loc, 4, nullptr, 0));
+                return ASRUtils::EXPR(ASR::make_IntegerBinOp_t(al, loc, left, op, right, type_, nullptr));
```
Comment (czgdp1807) [inline src/libasr/pass/pass_utils.cpp:426]: I am a bit surprised. We are creating `IntegerBinOp` for character type?

Comment (certik) [reply src/libasr/pass/pass_utils.cpp:426]: Good catch --- we should at least document in comments why that is so.

Comment (Thirumalai-Shaktivel) [reply src/libasr/pass/pass_utils.cpp:426]: As we do not have `StringBinOp` in ASR, I used `IntegerBInOp`. Also, The output of StringCompare is always a bool and the product of a bool can be an Integer. That's why I used `IntgerBinOp`.

Comment (Thirumalai-Shaktivel) [reply src/libasr/pass/pass_utils.cpp:426]:
```Clojure
(If
    (IntegerBinOp
        (StringCompare
            (StringConstant
                "a"
                (Character 1 1 () [])
            )
            LtE
            (Var 3 c)
            (Character 1 1 () [])
            ()
        )
        Mul
        (StringCompare
            (Var 3 c)
            LtE
            (StringConstant
                "j"
                (Character 1 1 () [])
            )
            (Character 1 1 () [])
            ()
        )
        (Integer 4 [])
        ()
    )

```

Comment (czgdp1807) [reply src/libasr/pass/pass_utils.cpp:426]: Why are using `Mul` for multiplying two booleans? I think equivalent of `Mul` for logical operands is `And` operation? So `LogicalBinOp` makes more sense?

Comment (Thirumalai-Shaktivel) [reply src/libasr/pass/pass_utils.cpp:426]: I didn't look deeply into this implementation. But, I had the same thought as you did.

File: integration_tests/case_07.f90 [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
-    end select
-
+    implicit none
+    call test_case('a')
+    call test_case('d')
+    call test_case('l')
+    call test_case('p')
+    call test_case('z')
+    call test_case('k')
+    call test_case('r')
```
Comment (czgdp1807) [inline integration_tests/case_07.f90:9]: Convert `test_case` into a function, make it return different values for different cases and then use error stops to make sure the correct case was executed.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: integration_tests/case_07.f90 (hunk 2 above)
Comment (Thirumalai-Shaktivel) [reply integration_tests/case_07.f90:9]: Yup!

**Review by certik — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review

**Pull request merged**